### PR TITLE
Decompile scenery_multiple_paint

### LINKED ISSUE #3695: Problen with saving scenery
**OS:** windows7
**Version:** v0.0.5
**Commit:** c43e3ee

Explanation of the issue...When trying to save a ride and the "ADD SCENERY" button is pressed you get some scenery - it is not all grayed out like it is supposed to be. Upon selecting the scenery the items already "selected" don't show up when saving. If you select them, they go dark. Selecting them again doesn't help. When saving the ride only the items you selected are saved. Also as per the second pic extra items show up.

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Multiplayer?

**Steps to reproduce:**
1. 
2. 

**Dump file**
If you have a dump file, zip it and drag&drop it here.

**Screenshots / Video:**
scr2
scr4

Drag / drop screenshots here. Use https://vid.me to upload video.

**Save game:**
Change the file extension to .txt or package to a .zip so that it can be drag / dropped here...

### PR DESCRIPTION
**Pull request by zsilencer**
also fixed #3695 and a bug with scrolling_text_get_colour

### CODE REVIEW

File: src/paint/map_element/scenery_multiple.c [hunk 1]
```diff
@@ -15,9 +15,314 @@
...
+	if (entry->large_scenery.flags & 0x40) {
+		paint_util_set_segment_support_height(SEGMENTS_ALL, clearanceHeight, 0x20);
+	} else {
+		paint_util_set_segment_support_height(SEGMENTS_ALL, 0xFFFF, 0);
+	}
+
+	if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) < clearanceHeight) {
+		RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) = clearanceHeight;
+		RCT2_GLOBAL(0x141E9DA, uint8) = 0x20;
+	}
```
Comment (IntelOrca) [inline src/paint/map_element/scenery_multiple.c]: Use `paint_util_set_general_support_height`

File: src/paint/map_element/scenery_multiple.c [hunk 2]
```diff
@@ -15,9 +15,314 @@
...
+}
+
+/*
+*
+* rct2: 0x006B7F0C
+*/
 void scenery_multiple_paint(uint8 direction, uint16 height, rct_map_element *mapElement) {
-	RCT2_CALLPROC_X(0x6B7F0C, 0, 0, direction, height, (int)mapElement, 0, 0);
+	//RCT2_CALLPROC_X(0x6B7F0C, 0, 0, direction, height, (int)mapElement, 0, 0); return;
+	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8_t) = VIEWPORT_INTERACTION_ITEM_LARGE_SCENERY;
```
Comment (IntelOrca) [inline src/paint/map_element/scenery_multiple.c]: Use `gPaintInteractionType`

File: src/paint/map_element/scenery_multiple.c [hunk 3]
```diff
@@ -15,9 +15,314 @@
...
+	if (RCT2_GLOBAL(0x009DEA6F, uint8) & 1) {
+		if (!track_design_save_contains_map_element(mapElement)) {
+			ebp = 0x21700000;
+			image_id &= 0x7FFFF;
+			dword_F4387C = ebp;
+			image_id |= dword_F4387C;
+		}
+	}
+	if (mapElement->flags & MAP_ELEMENT_FLAG_GHOST) {
+		RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8_t) = VIEWPORT_INTERACTION_ITEM_NONE;
```
Comment (IntelOrca) [inline src/paint/map_element/scenery_multiple.c]: Use `gPaintInteractionType`

**Comment by zsilencer** [comment]
there are a couple of small quirks that i would like to fix first so dont merge yet

**Comment by zsilencer** [comment]
OK it should be good now, the 3d signs are now pixel perfect with original

**Comment by Wirlie** [comment]
Hi! I tested your changes with the 3D signs and I have seen this bug:


This only occurrs if the 3D sign preview have a special character (like latin chars, for example: Ñ)

**Comment by zsilencer** [comment]
OK maybe you can try it again I've added the utf8 handling

**Comment by Wirlie** [comment]
Awesome! @zsilencer thank you very much, works fine for me. Issue #3086 can be closed after merge.

**Pull request merged**